docs/ProductionChecklistForScalarDLLedger.md

```diff
@@ -0,0 +1,156 @@
+
+#### Required for production environments
+
+```mermaid
+graph LR
+  subgraph "Administrative domain 1"
+    subgraph "Another environment"
+      A-1[User application]
+    end
+  end
+  subgraph "Administrative domain 2"
+    subgraph "Kubernetes cluster"
+      B-1[ScalarDL Ledger]
+    end
+  end
+  A-1 --> B-1
+```
+
+#### Not recommended for production environments (for testing purposes only)
+
+```mermaid
+graph LR
+  subgraph "Kubernetes cluster"
+    direction LR
+    A-1[User application] --> A-2[ScalarDL Ledger]
+  end
+```
+
+### Check if your contract and function follow guidelines
+
+You should check if your contract and function follow the guidelines:
+
+* [A Guide on How to Write a Good Contract for ScalarDL](https://github.com/scalar-labs/scalardl/blob/master/docs/how-to-write-contract.md)
+* [A Guide on How to Write Function for ScalarDL](https://github.com/scalar-labs/scalardl/blob/master/docs/how-to-write-function.md)
+
+### Versioning contracts
+
+When you register a contract once, you cannot overwrite the existing contract. So, you should consider the versioning of contracts. We recommend one of the following two methods:
+
+#### Versioning by using `Class Name`
+
+```console
+Contract ID              : FooV1
+Binary Name              : com.example.contract.FooV1
+Class file (Class Name)  : src/main/java/com/example/contract/FooV1.class
+---
+Contract ID              : FooV2
+Binary Name              : com.example.contract.FooV2
+Class file (Class Name)  : src/main/java/com/example/contract/FooV2.class
+```
+
+#### Versioning by using `Package Name`
+
+```console
+Contract ID             : FooV3
+Binary Name             : com.example.contract.v3.Foo
+Class file (Class Name) : src/main/java/com/example/contract/v3/Foo.class
+---
+Contract ID             : FooV4
+Binary Name             : com.example.contract.v4.Foo
+Class file (Class Name) : src/main/java/com/example/contract/v4/Foo.class
+```
+
+### Use the same binary name, package name, and class name when you register the contract
+
+If the binary name, package name, and class name are different when you register the contract, you cannot execute such a contract after you register it.
+
+#### Binary name and class name are different (you cannot execute this contract)
+
+```console
+Contract ID              : FooV5
+Binary Name              : com.example.contract.FooV5
+Class file (Class Name)  : src/main/java/com/example/contract/FooV6.class
+```
+
+#### Binary name and package name are different (you cannot execute this contract)
+
+```console
+Contract ID              : FooV7
+Binary Name              : com.example.contract.v7.Foo
+Class file (Class Name)  : src/main/java/com/example/contract/v8/Foo.class
+```
+
+### Secret key and certification file requirements
+
+You must make sure that your secret key and certification file match the following ScalarDL requirements.
```

Comment:
The comment [here](https://github.com/scalar-labs/scalar-kubernetes/pull/215#discussion_r1375703097) should also be applied to the above?

Reply:
Ah, yes. You are right.
I overlooked it...
Thank you for pointing it out!
I fixed it in d2857e8c9001f36ea0d57ef4e89ab3dfda2082b6.